src/Nethermind/Nethermind.Evm/ReadOnlyTransactionProcessor.cs

```diff
@@ -43,6 +46,7 @@ public void CallAndRestore(Transaction transaction, BlockHeader block, ITxTracer
 
         public void Dispose()
         {
+            _stateProvider.StateRoot = _stateBefore;
```

Comment:
slet us confirm there is no dirty state here (no Restore is called)

Reply:
Won't reset in next line handle that fine? Maybe I'm not seeing something.

Reply:
it is a different hing to reset state root and to Reset the potential dirty nodes on the trie